### PR TITLE
GH-1019 - Fix C3 Bar Clipping

### PR DESCRIPTION
**Pull request by mlzummo**
Fix C3 Bar Clipping
Fixes GH-1019

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/c3chart/Common.js [hunk 1]
```diff
@@ -88,6 +88,23 @@
...
 
+        var labelText = this.data().map(function(row) {
+            return row[0];
+        });
+
+        var tempNode = element.append("div").attr("class","c3 temp").append("svg").append("g").attr("class","c3-axis c3-axis-x");
+        tempNode.selectAll("g").data(labelText).enter().append("g").attr("class","tick").append("text").append("tspan").text(function(d) { return d; });
+
+        var xyz = d3.max(d3.selectAll('.c3-axis-x g.tick > text > tspan')[0].map(function(a) {
+            return a.getComputedTextLength(); // in future with rotation might need BBox
```
Comment (GordonSmith) [inline src/c3chart/Common.js]: I have a vague memory of not being able to use ```getComputedTextLength``` in IE or FF or suchlike...

**Comment by GordonSmith** [comment]
@jbrundage please review

**Comment by GordonSmith** [comment]
(lint failing)

**Comment by jbrundage** [comment]
@mlzummo This change doesn't allow the chart's left-padding to be re-evaluated if/when the data is changed.

Run this in the console and you'll see what I mean:
currWidget.data([["This is a test",75,68],["This is also a test",45,55]]).render()

**Comment by mlzummo** [comment]
@GordonSmith @jbrundage found a diff and better fix please re-review

**Comment by jbrundage** [comment]
Very nice... works like a charm :+1:

**Pull request merged**